microsoft/testsuites/core/debug.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from __future__ import annotations
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.sut_orchestrator import AZURE, READY
+from lisa.tools import Cat, Echo, KernelConfig, Mount
+from lisa.tools.mkfs import FileSystem
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers kernel debug functionalities.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class KernelDebug(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case check VM can be enabled kprobe.
+
+        Steps:
+        1. Check if CONFIG_KPROBE_EVENTS is enabled in kernel config.
+        2. Check if /sys/kernel/debug/tracing/ is mounted, if not, mount it.
+        3. Get origin values of /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable.
+        4. Write "p:my filp_close" to /sys/kernel/debug/tracing/kprobe_events and
+         write "1" to /sys/kernel/debug/tracing/events/kprobes/my/enable.
+        5. Check if /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable are changed.
+        6. Write origin values back to /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+    )
+    def verify_enable_kprobe(self, node: Node) -> None:
+        if not node.tools[KernelConfig].is_enabled("CONFIG_KPROBE_EVENTS"):
+            raise SkippedException("CONFIG_KPROBE_TRACING is not enabled")
+        else:
```

Comment:
The `else` is not necessary. When the `if` block is true, and exception happens, the following code won't be executed. Removing the `else` will reduce code complex.

Reply:
updated